horovod/runner/elastic/discovery.py

```diff
@@ -39,13 +52,48 @@ def get_event(self):
     def set_event(self):
         self._event.set()
 
+    def _in_cooldown_period(self, current_time):
+        return self._cooldown_period_end_ts > current_time
+
+
+    def _set_cooldown_period(self, current_time):
+        if self._cooldown_lower_limit == -1 or self._cooldown_upper_limit == -1:
+            return
+        self._blacklist_count += 1
+        def _exponential_backoff_time():
+            cooldown_delay = self._cooldown_lower_limit * (1 << self._blacklist_count)
+            + (random.uniform(0,1) * self._cooldown_lower_limit)
```

Comment:
Can you move the `+` to the previous line and intend the this line with the `self` above?
Also the brackets are redundant.

Does this code even work without a `\` at the end of the previous line?

Reply:
Done